strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseController.java

```diff
@@ -174,4 +171,277 @@
+            sb.append("<head>");
+            sb.append(
+                    "<style>body{font-family: \"Trebuchet MS\", verdana, lucida, arial, helvetica, sans-serif;} table tr {text-align: left;}</style>");
+            sb.append("<title>Index of " + request.getRequestURI() + "</title>");
+            sb.append("</head>");
+            sb.append("<body>");
+            sb.append("<h1>Index of " + request.getRequestURI() + "</h1>");
+            sb.append("<table cellspacing=\"10\">");
+            sb.append("<tr>");
+            sb.append("<th>Name</th>");
+            sb.append("<th>Last modified</th>");
+            sb.append("<th>Size</th>");
+            sb.append("<th>Description</th>");
+            sb.append("</tr>");
+            sb.append("<tr>");
+            sb.append("<td colspan=4><a href=\"..\">..</a></td>");
+            sb.append("</tr>");
+           
+            for (FileContent fileContent : content.getDirectories())
+            {
+                appendFile(sb, fileContent, requestUri, true, url);
+            }
+                        
+            if(url == null && !content.getFiles().isEmpty())
+            {
+                logger.debug("Files are present outside a repository");
+                response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+                return;
+            }
+            
+            for (FileContent fileContent : content.getFiles())
+            {
+                appendFile(sb, fileContent, requestUri, false, url);
+            }
+
+
+            sb.append("</table>");
+            sb.append("</body>");
+            sb.append("</html>");
+
+            response.setContentType("text/html;charset=UTF-8");
+            response.setStatus(HttpStatus.OK.value());
+            response.getWriter()
+                    .write(sb.toString());
+            response.getWriter()
+                    .flush();
+            response.getWriter()
+                    .close();
+
+        }
+        catch (IOException e)
+        {
+            logger.error(" error accessing requested directory");
+
+            response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+        }
+    }
+    
+    private boolean appendFile(StringBuilder sb,
+                               FileContent fileContent,
+                               String requestUri,
+                               boolean isDirectory,
+                               URL url)
+            throws IOException
+    {
+        String name = fileContent.getName();
+        String size = fileContent.getSize();
+        String lastModified = fileContent.getLastModified();
+
+        sb.append("<tr>");
+
+        if(isDirectory)
+        {
+            sb.append("<td><a href='" + URLEncoder.encode(name, "UTF-8") + 
+                      "/'>" + name + "/" + "</a></td>");
+        }
+        else
+        {  
+            String storagesUrl = url.toString() + "/" + URLEncoder.encode(name, "UTF-8");
```

Comment:
@sbespalov 2. Used the URL here for getting the URL pointing to /storages

Reply:
I think `artefactUrl` will be more meaningfully instead of `storagesUrl`.
Am I understand correctly that this is URL to download artefact?

Reply:
Yes you are right!

Reply:
also, why you concatenate `name` to URL?
this URL assumed to be ready to use without modifications

Reply:
Because URL is URL for current folder browse request  which only points to parent folder of files. So I resolve path for parent folder and append filename.

Reply:
the main thing of `artifactResolutionService.resolveResource` method is to avoid additional modifications for URL you get from it, so I think in case of artifact files you need to construct URL just with this method without these modifications